### PR TITLE
update h2 to v2 and fix schema creation for it

### PR DESCRIPTION
**Pull request by aldenml**
This address https://github.com/akka/akka-persistence-jdbc/issues/639 and https://github.com/akka/akka-persistence-jdbc/pull/633

### CODE REVIEW

**Comment by aldenml** [comment]
Ok, I need to fix `H2JournalMigratorTest`

**Review by octonato — COMMENTED** [review]
Thanks @aldenml. It's looking good, but I have one question. Please, see below.

File: core/src/main/resources/schema/h2/h2-create-schema.sql [hunk 1]
```diff
@@ -39,8 +39,10 @@ CREATE TABLE IF NOT EXISTS "snapshot" (
     PRIMARY KEY("persistence_id","sequence_number")
     );
 
+CREATE SEQUENCE IF NOT EXISTS "global_offset_seq";
+
 CREATE TABLE IF NOT EXISTS "durable_state" (
-    "global_offset" BIGINT NOT NULL AUTO_INCREMENT,
+    "global_offset" BIGINT DEFAULT NEXT VALUE FOR "global_offset_seq",
```
[the remark this answers is not in the document]
Comment (aldenml) [reply core/src/main/resources/schema/h2/h2-create-schema.sql:45]: Hi @octonato, yes, the problem is with the implementation of `H2SequenceNextValUpdater#getSequenceNextValueExpr`  
(https://github.com/aldenml/akka-persistence-jdbc/blob/master/core/src/main/scala/akka/persistence/jdbc/state/SequenceNextValUpdater.scala#L31). Apparently, there is a difference in what's created behind the scenes in H2 to support the "auto_increment". In order to keep the same code in Scala, I decided to use a user-created `SEQUENCE `.

**Review by octonato — APPROVED** [review]
LGTM

However, I realise now that we will need to add some notice in the release notes. I previously said that it's ok  to change the schema for H2 because we only recommend it for testing. 

But even for testing, users may need to adapt their code. A user that is creating some tables during test may need to adapt their SQL because of H2 v2. That's still fine, we just need to make it clear upfront. 

I will summoned @patriknw and @johanandren to get their opinions on this. 

My take, is that we should move on with the upgrade of H2. We were using a quite old version and this is not supposed to be a production one.

**Comment by johanandren** [comment]
I don't see that we clearly express that it is only for testing in the docs, so users may use it for prod even though it is a bad idea (tm). I think there needs to be some docs about how migration can be done.

**Comment by octonato** [comment]
> I don't see that we clearly express that it is only for testing in the docs, so users may use it for prod even though it is a bad idea (tm). I think there needs to be some docs about how migration can be done.

I really thought we were very explicit about it, but apparent not. I also checked Akka Projections and we don't mention it either. 

So, indeed, we will need to add docs about migration.

**Comment by aldenml** [comment]
is this additional documentation something that should be done as part of this PR?

**Comment by octonato** [comment]
> is this additional documentation something that should be done as part of this PR?

No, we can do it in another PR. The only problem is that we should not merge this one before we are ready with the migration notice. 
If we merge and it happens that we need to cut a release because of a bug, we will impact the users directly and without notice. 

If you want to add the migration notes to this PR, you are welcome. Otherwise, we keep it around until someone (we or community) has the bandwidth to do it.

**Comment by octonato** [comment]
I haven't said it before, so here it is. 😄 

Thanks for the contribution. I hope we can get the migration guide soon and move forward with this PR.

**Comment by nvollmar** [comment]
We're also waiting to upgrade, is the only thing missing the migration notes?

**Review by johanandren — APPROVED** [review]
LGTM, let's follow up with migration notes

**Pull request merged**